extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Extending Materials
+
+The PBR clearcoat materials are defined by adding the `KHR_materials_clearcoat` extension to any compatible glTF material (excluding those listed above).  For example, the following defines a material like varnish using clearcoat parameters.
+
+```json
+{
+    "materials": [
+        {
+            "name": "varnish",
+            "extensions": {
+                "KHR_materials_clearcoat": {
+                    "clearcoatFactor": 1.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Clearcoat
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatFactor**               | `number`                                                                        | The clearcoat layer intensity.         | No, default: `0.0`   |
+|**clearcoatTexture**              | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer intensity texture. | No                   |
+|**clearcoatRoughnessFactor**      | `number`                                                                        | The clearcoat layer roughness.         | No, default: `0.0`   |
+|**clearcoatRoughnessTexture**     | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer roughness texture. | No                   |
+|**clearcoatNormalTexture**        | [`normalTextureInfo`](/specification/2.0/README.md#reference-normaltextureinfo) | The clearcoat normal map texture.      | No                   |
+  
+The clearcoat formula `f_clearcoat` is computed using the specular term from the glTF 2.0 Metallic-Roughness material, defined in [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation).  Use F0 equal `0.04`, base color white `1.0, 1.0, 1.0`, metallic value `0.0`, and the clearcoat roughness value defined in this extension as follows:
+
+```
+clearcoatRoughness = clearcoatRoughnessFactor * clearcoatRoughnessTexture.g
+```
+
+The following abstract code describes how the base and clearcoat layers should be blended together:
+  
+```
+clearcoatWeight = clearcoatTexture.r * clearcoatFactor * fresnel(0.04, NdotV)
```

Comment:
During code review, it was found I was passing `NdotV` into a `fresnel` function that wants to take `VdotH` as a parameter.  Which is correct?  It looks like the paper says `VdotH` goes here, not `NdotV`.

@UX3D-nopper Do you know where this term comes from?

Reply:
/cc @donmccurdy @bghgary @sebavan

Reply:
Good, that you review this.
To answer your question, NdotV is used e.g. in Subatsance Painter.
Furthermore, I would prefer following formula:
`color = (f_emissive + f_diffuse) * (1.0 - clearcoatBlendFactor) + mix(f_specular, f_clearcoat, clearcoatBlendFactor);`
Which paper are you refering to?

Reply:
In eevee from Blender, they also use NdotV (see lit_surface_frag.glsl).

Reply:
E.g. if you compare with Enterprise PBR:
https://dassaultsystemes-technology.github.io/EnterprisePBRShadingModel/spec-2020x.md.html#components/clearcoat
They are using VdotH but then you also need to blend with NdotV and NdotL. In this model, one has to calculate three fresnel terms. As far as I can remember, this was considered to be too expensive. This is the reason, for real-time rendering, NdotL is used in most cases for approximation. However, the blending currently defined is not used in Blender eevee and/or Substance Painter.

Reply:
The reason I'm asking about `NdotV` here, is because in [Appendix B](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#appendix-b-brdf-implementation) we specify `VdotH` as the parameter for the main Fresnel Schlick (F) for the core material, and we're referencing that appendix here.  I tried rendering with `VdotH` and the results look very similar, actually a little softer on the coating and more of the base layer visible, with less "glowing" at the edges.  I guess I'm OK with either one, I just wanted to double-check this since we do site Appendix B and then promptly use a different term from the one given there.

Reply:
OK, keeping `NdotV` makes sense.  For those using IBL exclusively, there won't be an `H` at all.